doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,9 @@
 
 <h3>New features since last release</h3>
 
+* Function is added for user-facing circuit decomposition. 
```

Comment:
We can probably come back to this later in the release cycle, but we may want to flesh this entry out a bit.  
```suggestion
* `qml.transforms.decompose` is added for stepping through decompositions to a target gateset.
```

We may want to also include an example, but we can leave that up to the final pre-release changelog cleaning.